transcom/mymove
# B 18902 SH to LH and LH to SH INT

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/TeamRoom.mvc/Show/808731)

## Summary

Additional logic in the short haul to line haul and line haul to short haul update on a shipment destination address change.

To test shorthaul to linehaul situations in general:
1. Create a local move with the same zip for current and destination zips, explained more below. Then change the destination zip to something further than 50 miles.

To test linehaul to shorthaul situations in general:
1. Create a move with the different current and destination zips. Then change the destination zip to match the current zip.

Situations where the original DLH or DSH service item should not be rejected and a new one should not be created:
1. A payment request exists on the DLH or DSH and is in approved, paid, or sent to gex state.
2. Destination SIT is present on the shipment, and any one of the four service items are in an Approved state.

Otherwise, you should see the DLH or DSH service item rejected and a new DSH or DLH one created. 

To test shorthaul to linehaul for when origin SIT is requested but not yet approved:
1. Create a LOCAL move (set the orders type to local move) as the customer, keeping track of your zip codes as you go.
4. Review as a services counselor and submit the move.
5. Switch over to the TOO role. Approve the shipment and add move management service items to the shipment, selecting "Approve and send".
6. You should see the move in the prime sim now. Check the database for the service items and verify you have 6 service items and one of those is DSH, domestic shorthaul.
7. Request origin SIT, but do not approve any of the service items as the TOO yet.
8. Make a note of your origin zip code. While the origin SIT service items are all still in the submitted state, make a shipment destination address change over 50 miles to trigger going from shorthaul to linehaul. 40356 -> 67492
9. Review and approve the shipment address change as TOO.
10. Check the database or MTO page and verify that the DSH service is now rejected with a rejection reason listed of “Automatically rejected due to change in destination address affecting the ZIP code qualification for short haul / line haul.”
11. Check the database or MTO page and verify that a DLH service item now exists, and is in approved status.

To test shorthaul to linehaul for when origin SIT is requested and approved:
1. Follows steps 1-4 from above.
3. Approve any or all of the service items.
4. Follow steps 6-9 from above.

To test linehaul to shorthaul for origin:
1. Create a non local move as the customer, keeping track of your zip codes as you go. Curr duty location: 40356, new duty location: 67492. Move: GPDYVX 
2. Review as a services counselor and submit the move.
3. Switch over to the TOO role. Approve the shipment and add move management service items to the shipment, selecting "Approve and send".
5. You should see the move in the prime sim now. Check the database for the service items and verify you have 6 service items and one of those is DLH, domestic linehaul.
6. Request origin SIT, but do not approve any of the service items as the TOO yet.
7. Make a note of your origin zip code. While the origin SIT service items are all still in the submitted state, make a shipment destination address change to the same zip as your original/curr duty location to trigger going from linehaul to shorthaul. 67492 -> 40356
8. Review and approve the shipment address change as TOO.
9. Check the database or MTO page and verify that the DLH service is now rejected with a rejection reason listed of “Automatically rejected due to change in destination address affecting the ZIP code qualification for short haul / line haul.”
10. Check the database or MTO page and verify that a DSH service item now exists, and is in approved status.

To linehaul to shorthaul for origin for when origin SIT is requested and approved:
1. Follows steps 1-4 from above.
2. Approve any or all of the service items.
3. Follow steps 6-9 from above.

To test linehaul to shorthaul for destination when destination is requested but not yet approved:
1. Create a non local move as the customer, keeping track of your zip codes as you go. 
2. Review as a services counselor and submit the move.
3. Switch over to the TOO role. Approve the shipment and add move management service items to the shipment, selecting "Approve and send".
4. You should see the move in the prime sim now. Check the database for the service items and verify you have 6 service items and one of those is DLH, domestic linehaul.
5. Request destination SIT, but do not approve any of the service items as the TOO yet.
6. Make a note of your origin zip code. While the destination SIT service items are all still in the submitted state, make a shipment destination address change to the same zip as your original/curr duty location to trigger going from linehaul to shorthaul.
7. Review and approve the shipment address change as TOO.
8. Check the database or MTO page and verify that the DLH service is now rejected with a rejection reason listed of “Automatically rejected due to change in destination address affecting the ZIP code qualification for short haul / line haul.”
9. Check the database or MTO page and verify that a DSH service item now exists, and is in approved status.
